src/shared/src/types.rs

```diff
@@ -54,10 +54,12 @@ pub mod custom_token {
     use candid::{CandidType, Deserialize, Principal};
 
     pub type LedgerId = Principal;
+    pub type IndexId = Principal;
```

Comment:
It is possible to wrap this in a so-called phantom type, that will prevent a `LedgerId` from being used where an `IndexId` is intended.  However I don't know what the consequences are of doing this on a public type.  E.g. does `didc` understand phantom types and pass them through transparently?  As such I consider this as something nice to test and roll out at some time in future.  I don't see that the risks from type confusion here are high enough to warrant doing that before we have product market fit.

Reply:
I'm not familiar with this pattern. I had a quick look to the IC repo and it does not seem to be used that frequently so, probably requires more know-how that mine. I'll merge the PR and let you have a look afterwards if you don't mind?

It's not production nor staging yet, so we can iterate this in another PR.